src/data/action.ts

```diff
@@ -76,6 +84,7 @@ export function action<T extends Array<any>, U = void>(
     function handler(error?: boolean) {
       return async (res: any) => {
         const result = await handleResponse(res, error, router.navigatorFactory());
+        o.onComplete && o.onComplete(submission);
```

Comment:
So I have already been playing with this `onComplete` hook a bit. Bleeding edge I know lol

I think this should actually be moved down a couple lines to be after you've `setResult(result)`. Otherwise the submission.result is always undefined in the `onComplete` handler

Reply:
Good call.. Hmm.. it's tricky because when we set the result we trigger reactivity. I guess I should make a copy of the submission object that overrides that property value. I think timing wise we want to run `onComplete` before we actually render anything.

Reply:
Ok I think I solved it with the latest version I just pushed. We make a clone and then intercept the retry operation to ensure that it works as expected.